partition_table.go

```diff
@@ -667,7 +667,7 @@ func (p *PartitionTable) CurrentState() PartitionStatus {
 }
 
 // WaitRecovered returns a channel that closes when the partition table enters state `PartitionRunning`
-func (p *PartitionTable) WaitRecovered() chan struct{} {
+func (p *PartitionTable) WaitRecovered() <-chan struct{} {
 	return p.state.WaitForState(State(PartitionRunning))
 }
 
```

Comment:
IMO this should be a read only channel.
But It's a breaking change. 
Don't know if we should keep it.

Reply:
it's breaking but it fixes a bug so we definitely have to change that. If someone deliberately closed the channel from the outside it was wrong anyway :)